blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/ArtifactContainerImpl.java

```diff
@@ -0,0 +1,91 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+import com.google.common.util.concurrent.Service;
+import hudson.model.Run;
+import io.jenkins.blueocean.commons.ServiceException;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BlueArtifact;
+import io.jenkins.blueocean.rest.model.BlueArtifactContainer;
+import jenkins.util.VirtualFile;
+import org.kohsuke.stapler.Stapler;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+public class ArtifactContainerImpl extends BlueArtifactContainer {
+    final private Run run;
+    final private Link self;
+    public ArtifactContainerImpl(Run r, Reachable parent) {
+        this.run = r;
+        this.self = parent.getLink().rel("artifacts");
+    }
+
+
+    @Override
+    public Link getLink() {
+        return self;
+    }
+
+    @Override
+    public BlueArtifact get(final String name) {
+        final VirtualFile file = run.getArtifactManager().root().child(name);
+        try {
+            if(file == null || !file.exists() || !file.isFile()) {
+                return null;
+            }
+        } catch (IOException e) {
+            throw new ServiceException.UnexpectedErrorException("Something is wrong with an artifact",e);
+        }
+        return new BlueArtifact() {
+            @Override
+            public String getName() {
+                return file.getName();
+            }
+
+            @Override
+            public String getPath() {
+                return name;
+            }
+
+            @Override
+            public String getUrl() {
+                return Stapler.getCurrentRequest().getContextPath() +
+                    "/" + run.getUrl()+"artifact/"+ name;
+            }
+
+            @Override
+            public long getSize() {
+                try {
+                    return file.length();
+                } catch (IOException e) {
+                    throw new ServiceException.UnexpectedErrorException("qError getting file length", e);
+                }
+            }
+
+            @Override
+            public Link getLink() {
+                return new Link(getUrl());
+            }
+        };
+    }
+
+    @Override
+    public Iterator<BlueArtifact> iterator() {
```

Comment:
Default implementation of iterator() should not hardcode to 100.  Instead implement iterator(start, limit), this way pagination will be enforced early on for clients calling with start and limit parameters.